components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -8679,8 +8750,20 @@ public void deleteSharedScope(String scopeName, String tenantDomain) throws APIM
         if (log.isDebugEnabled()) {
             log.debug("Deleting shared scope " + scopeName);
         }
-        KeyManagerHolder.getKeyManagerInstance(tenantDomain).deleteScope(scopeName);
-        ApiMgtDAO.getInstance().deleteSharedScope(scopeName, tenantDomain);
+        Map<String, KeyManagerDto> tenantKeyManagers = KeyManagerHolder.getTenantKeyManagers(tenantDomain);
+        for (Map.Entry<String,KeyManagerDto> keyManagerEntry : tenantKeyManagers.entrySet()){
+            KeyManager keyManager = keyManagerEntry.getValue().getKeyManager();
+            if (keyManager != null){
+                try{
+                    keyManager.deleteScope(scopeName);
+                }catch (APIManagementException e){
+                    log.error("Error while Deleting Shared Scope " + scopeName + " from KeyManager " +
+                            keyManagerEntry.getKey(), e);
```

Comment:
same here. formatting issues.

Shouldnt we throw the exception?

Reply:
no, since we have multiple key managers we only log the exception.

@bhathiya  in here if we store key manager which we publish scope. we can iterate only those key managers and remove it.
wdyt ?